src/charm.py

```diff
@@ -4,19 +4,169 @@
+        return [
+            NetworkAttachmentDefinition(
+                metadata=ObjectMeta(name=CORE_GW_NAD_NAME),
+                spec={
+                    "config": json.dumps(
+                        {
+                            "cniVersion": "0.3.1",
+                            "type": "macvlan",
+                            "ipam": {
+                                "type": "static",
+                                "routes": [
+                                    {
+                                        "dst": self._get_ue_subnet_config(),
+                                        "gw": self._get_upf_core_ip_config(),
+                                    }
+                                ],
+                                "addresses": [
+                                    {
+                                        "address": self._get_core_gateway_ip_config(),
+                                    }
+                                ],
+                            },
+                            "capabilities": {"mac": True},
+                        }
+                    )
+                },
+            ),
+            NetworkAttachmentDefinition(
+                metadata=ObjectMeta(name=RAN_GW_NAD_NAME),
+                spec={
+                    "config": json.dumps(
+                        {
+                            "cniVersion": "0.3.1",
+                            "type": "macvlan",
+                            "ipam": {
+                                "type": "static",
+                                "addresses": [
+                                    {
+                                        "address": self._get_ran_gateway_ip_config(),
+                                    }
+                                ],
+                            },
+                            "capabilities": {"mac": True},
+                        }
+                    )
+                },
+            ),
+            NetworkAttachmentDefinition(
+                metadata=ObjectMeta(name=ACCESS_GW_NAD_NAME),
+                spec={
+                    "config": json.dumps(
+                        {
+                            "cniVersion": "0.3.1",
+                            "type": "macvlan",
+                            "ipam": {
+                                "type": "static",
+                                "addresses": [
+                                    {
+                                        "address": self._get_access_gateway_ip_config(),
+                                    }
+                                ],
+                            },
+                            "capabilities": {"mac": True},
+                        }
+                    )
+                },
+            ),
+        ]
+
+    def _on_config_changed(self, event: EventBase) -> None:
+        """Config changed event."""
+        if not self._container.can_connect():
```

Comment:
should we check for invalid configs as in the other charms?

Reply:
Agreed, done